src/pke/lib/scheme/bfvrns/bfvrns-parametergeneration.cpp

```diff
@@ -62,7 +62,7 @@ bool ParameterGenerationBFVRNS::ParamsGenBFVRNS(std::shared_ptr<CryptoParameters
     MultiplicationTechnique multTech = cryptoParamsBFVRNS->GetMultiplicationTechnique();
     ProxyReEncryptionMode PREMode    = cryptoParamsBFVRNS->GetPREMode();
 
-    if (PREMode != INDCPA) {
+    if ((PREMode != INDCPA) && (PREMode != NOT_SET)) {
         OPENFHE_THROW(not_available_error, "HRA modes for PRE with BFVRNS not implemented yet");
```

Comment:
i would throw an exception with a different error message which includes the value of PREMode. something like this:
"this ProxyReEncryptionMode " + PREMode + " is not supported for BFVRNS". The same should apply to
src/pke/lib/scheme/bgvrns/bgvrns-parametergeneration.cpp: l.334
src/pke/lib/scheme/ckksrns/ckksrns-parametergeneration.cpp: l.62